libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -248,8 +232,12 @@ MAV_RESULT AP_Camera::handle_command_long(const mavlink_command_long_t &packet)
         }
         return MAV_RESULT_ACCEPTED;
     case MAV_CMD_SET_CAMERA_ZOOM:
-        if (is_equal(packet.param1, (float)ZOOM_TYPE_CONTINUOUS)) {
-            set_zoom_step((int8_t)packet.param2);
+        if (is_equal(packet.param1, (float)ZOOM_TYPE_CONTINUOUS) &&
+            set_zoom(AP_Camera::ZoomType::RATE, packet.param2)) {
```

Comment:
```suggestion
            set_zoom(AP_Camera::ZoomType::INCREMENT, packet.param2)) {
```

Reply:
Glad you like it and thanks for the review.  I think that "RATE" is more accurate than INCREMENT because, for me at least, increment implies that the zoom multiple (e.g. 5x, 10x) is being increased by 1 (e.g. 5x -> 6x) but that's not actually what this control does.  It actually starts zoom in or out at some (camera specific) rate.

Reply:
> Glad you like it and thanks for the review. I think that "RATE" is more accurate than INCREMENT because, for me at least, increment implies that the zoom multiple (e.g. 5x, 10x) is being increased by 1 (e.g. 5x -> 6x) but that's not actually what this control does. It actually starts zoom in or out at some (camera specific) rate.

So the camera continues to change zoom level indefinitely after we process just one of these messages?!  That's terrible!

Reply:
@peterbarker,

Yes, that's right although of course, it has a physical limit so it will eventually stop at either completed zoomed-in or completely zoomed-out.  In general this control is connected to an auxiliary switch which is edge activated so it works.

I think we stumbled into this implementation though so if you've got suggestions on how to make it better I'm all ears!